### ISSUE TITLE
[DSMR] DSMR 5.0 unknown message (Voltage...?)

### ISSUE THREAD

**Issue by martinvw** [openhab/openhab#4792]
## Current Behavior
It seems that my recently installed DSMR meter returns some unknown message and it does so each second so it causes a lot of logging (which can of course be supressed)

```
2016-11-17 20:11:38.397 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:32.7.0, obis values:[226.6*V]
2016-11-17 20:11:38.398 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:52.7.0
2016-11-17 20:11:38.398 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:52.7.0, obis values:[224.7*V]
2016-11-17 20:11:38.504 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:32.7.0
2016-11-17 20:11:38.505 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:32.7.0, obis values:[226.7*V]
2016-11-17 20:11:38.506 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:52.7.0
2016-11-17 20:11:38.506 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:52.7.0, obis values:[224.6*V]
2016-11-17 20:11:38.507 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:72.7.0
2016-11-17 20:11:38.507 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:72.7.0, obis values:[229.7*V]
2016-11-17 20:11:38.619 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:32.7.0
2016-11-17 20:11:38.620 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:32.7.0, obis values:[226.5*V]
2016-11-17 20:11:38.620 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:52.7.0
2016-11-17 20:11:38.621 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:52.7.0, obis values:[224.7*V]
2016-11-17 20:11:38.622 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:72.7.0
2016-11-17 20:11:38.622 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:72.7.0, obis values:[229.6*V]
2016-11-17 20:11:38.726 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:32.7.0
2016-11-17 20:11:38.727 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:32.7.0, obis values:[226.7*V]
2016-11-17 20:11:38.728 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:52.7.0
2016-11-17 20:11:38.728 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:52.7.0, obis values:[224.8*V]
2016-11-17 20:11:38.729 [WARN ] [.b.d.i.messages.OBISMsgFactory] - Received OBIS unknown message:1-0:72.7.0
2016-11-17 20:11:38.729 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:72.7.0, obis values:[229.7*V]
```

## Possible Solution
Support should / could be added

## Your Environment
* I have a ISKRA AM550
* I run OH 1.8.3

**Comment by martinvw** [comment]
@mvolaart will it be as simple as adding the following fragment to the OBISMsgType:

```java
    EMETER_INSTANT_VOLTAGE_L1(new OBISIdentifier(1, 0, 32, 7, 0, null), DSMRMeterType.ELECTRICITY,
            new CosemValueDescriptor(CosemFloat.class, "V", "eInstantVoltageL1"),
            "Instantenous voltage on L1"),
    EMETER_INSTANT_VOLTAGE_L2(new OBISIdentifier(1, 0, 52, 7, 0, null), DSMRMeterType.ELECTRICITY,
            new CosemValueDescriptor(CosemFloat.class, "V", "eInstantVoltageL2"),
            "Instantenous voltage on L2"),
    EMETER_INSTANT_VOLTAGE_L3(new OBISIdentifier(1, 0, 72, 7, 0, null), DSMRMeterType.ELECTRICITY,
            new CosemValueDescriptor(CosemFloat.class, "V", "eInstantVoltageL3"),
            "Instantenous voltage on L3"),
```

Additional questions:

1. Do you have some kind of spec which proves that this indeed according to the standard for 5.0
2. Are you okay with the naming of the enum, itemId and the text?

**Comment by mvolaart** [comment]
@martinvw What a coincidence. I received my DSMR 5.0 meter yesterday and i saw the same error logging.

The answers to your questions:
1. Yes i have a specification and i can confirm the messages in the log are valid DSMR 5.0 messages (see http://www.netbeheernederland.nl/themas/hotspot/?dossierid=11010056&title=Slimme+meter for the specification document)
2. As far as i can see the suggested fix is correct (enum, itemId and the text).

**Comment by martinvw** [comment]
I also see one other probleem in my log, I fear that processing the data takes too long and that new data arrives before the previous data is consumed, is see this on a raspberry pi 2, do you also see this occurring?

```
2016-11-18 09:45:30.730 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Unexpected character '/' in state:DATA_OBIS_VALUE_END, header = /ISK5\2M550T-1011, obisId = 1-0:72.7.0, obisValue = 228.6*V. Publishing partial P1Telegram and wait for new P1Telegram
2016-11-18 09:45:36.273 [ERROR] [.b.d.i.messages.OBISMsgFactory] - Failed to parse OBIS identifier 1-0:72.7.0, values:[228
/ISK5\2M550T-1011

1-3:0.2.8(50]
2016-11-18 09:45:36.274 [WARN ] [o.o.b.d.i.p.P1TelegramParser  ] - Failed to parse:OBISDataLine [OBIS id:1-0:72.7.0, obis values:[228
/ISK5\2M550T-1011

1-3:0.2.8(50]
```

**Comment by martinvw** [comment]
I see/saw this occurring before and after I patched my dsmr binding.

**Comment by mvolaart** [comment]
Based on your last information it looks like data is may be overwritten. It is hard to tell at this point where in the system the data is lost (hardware, kernel, OpenHAB).
However based on your first message i see that within 1 second the same data is processed. I looked in my event.log and i see that with my old V3 meter i received one message each 10 seconds. With my new V5 meter i receive new data each 5 seconds, however each 5 seconds i see that the same data is processed 5x (like it is received 5x). This is unexpected and could be the cause of overflowing buffers in the raspberry (i have here a full server which manages to proces it in time).

I will investigate this further

**Comment by mvolaart** [comment]
I found the problem. DSMR V5 specifies an update interval of 1 second. Since the previous DSMR had an interval of 10 seconds specified the binding is only activated once every 5 seconds. Resulting in getting 5 messages in my setup and could also cause buffer overflow on the raspberry due to limited serial buffers.

I will make a fix for this.

**Comment by mvolaart** [comment]
Can you try this test version of the binding?

It should solve the problem. In my setup it is running correctly

[org.openhab.binding.dsmr-1.9.0-SNAPSHOT.zip](https://github.com/openhab/openhab/files/602288/org.openhab.binding.dsmr-1.9.0-SNAPSHOT.zip)

**Comment by martinvw** [comment]
Looking good, the updates are way more fluent now, I didn't yet notice that the data was processed in bulks, in OpenHab it looks also better. The only problem is that the graphs for my power usage are really unusable because they keep refreshing, each second. Any suggestions for that too?

**Comment by mvolaart** [comment]
I recognize the effect you describe, but i don't know a fix for this.

**Comment by martinvw** [comment]
Still going strong here 👍

**Comment by watou** [comment]
Closed via #4854
